### DISCUSSION TITLE
How to use stickers in messages?

### DISCUSSION THREAD

**Discussion by ThibaultJanBeyer** [AsamK/signal-cli#1306] [Q&A] [answered]
Hi all, I’m trying to send stickers, but I’m struggling with it:

1. How do get the sticker pack ID and the sticker ID?
I assumed that this can be done via signalstickers.com/ for example for https://signalstickers.com/pack/bf012b718285db8e3e86ceebddde3031 I assume that the ID is `bf012b718285db8e3e86ceebddde3031` OK 

2. Sticker pack not found
So I installed https://signalstickers.com/pack/bf012b718285db8e3e86ceebddde3031 on my account and am trying to use the CLI with `bf012b718285db8e3e86ceebddde3031:1` to send the first sticker of that pack. Unfortunately, it always responds with an error message:

```
INFO  AccountHelper - The Signal protocol expects that incoming messages are regularly received.
Failed to send sticker: Sticker pack not found
```

How can this be? What am I doing wrong?

Thank you!

**Comment by ThibaultJanBeyer** [comment]
Ah I think I found it!
The sticker IDs can be found here I think: https://github.com/signalstickers/stickers/blob/master/src/stickers.yml
This IS found on https://signalstickers.com/pack/bf012b718285db8e3e86ceebddde3031 IS the correct ID however the stickers I had installed had **a different ID**!

So basically you can disregard this question, and I’ll open a feature request ticket to get a list of all available sticker packs!

Thank you!

**Comment by ThibaultJanBeyer** [comment]
Ah no nevermind, the issue is that the sticker pack **has to be received once first** before it can be used :O

**Comment by AsamK** [comment, marked as answer]
There's a `listStickerPacks` command that shows all locally known sticker packs (it was just missing from the documentation).
I've also added a new `addStickerPack` command, that allows adding new sticker packs directly in signal-cli (with https://signal.art/addstickers/#pack_id=XXX&pack_key=XXX URLs)

**Reply by ThibaultJanBeyer** [reply]
Thank you, will give it a try shortly!
